### PR TITLE
docs: overhaul doc site and move component readmes to src

### PR DESCRIPTION
**Pull request by abutterworth**
A major update to the doc site. The design of pages and navigation is significantly changed.

Component documentation is now colocated with component source files. The doc site scans the src directory for markdown to parse.

image

### CODE REVIEW

File: scss/core/_typography.scss [hunk 1]
```diff
@@ -1,5 +1,10 @@
 @import "~bootstrap/scss/type";
 
+body {
```
Comment (davidjoy) [inline scss/core/_typography.scss:3]: This is going to change our actual stylesheet in apps, right?

File: www/LICENSE [hunk 2]
```diff
@@ -0,0 +1,14 @@
+The BSD Zero Clause License (0BSD)
```
Comment (davidjoy) [inline www/LICENSE:1]: Is this the correct license for us to use?

File: www/gatsby-config.js [hunk 3]
```diff
@@ -1,54 +1,73 @@
-const path = require('path');
+const path = require("path")
```
Comment (davidjoy) [inline www/gatsby-config.js:1]: wat!

File: scss/core/_typography.scss (hunk 1 above)
Comment (abutterworth) [reply scss/core/_typography.scss:3]: Yes. Good catch, I should move this to the doc site css and release this change separately if desired.

File: www/LICENSE (hunk 2 above)
Comment (abutterworth) [reply www/LICENSE:1]: nope.

File: www/gatsby-config.js (hunk 3 above)
Comment (abutterworth) [reply www/gatsby-config.js:1]: unintentional

File: www/src/components/Menu.jsx [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+                Component status
+              </Link>
+            </li>
+            <li>
+              <Link to="/insights">
+                Usage insights
+              </Link>
+            </li>
+            <li>
+              <a
```
Comment (davidjoy) [inline www/src/components/Menu.jsx:127]: Huh, when I ran the docs site locally, this <a> tag is a button, and it's non-functional.

**Review by davidjoy — APPROVED** [review]
The "Design Documentation" button doesn't seem to go anywhere.

**Pull request merged**